plugin/db/mysql/dump.go

```diff
@@ -226,26 +234,57 @@ func dumpTxn(ctx context.Context, txn *sql.Tx, database string, out io.Writer, s
 			}
 		}
 
+		// Disable foreign key check.
+		// mysqldump uses the same mechanism. When there is any schema or data dependency, we have to disable
+		// the unique and foreign key check so that the restoring will not fail.
+		if _, err := io.WriteString(out, disableUniqueAndForeignKeyCheckStmt); err != nil {
+			return err
+		}
+
 		// Table and view statement.
+		// We have to dump the table before views because of the structure dependency.
 		tables, err := getTablesTx(txn, dbName)
 		if err != nil {
 			return errors.Wrapf(err, "failed to get tables of database %q", dbName)
 		}
+		// Construct temporal views.
+		for _, tbl := range tables {
+			if tbl.TableType != viewTableType {
+				continue
+			}
+			if _, err := io.WriteString(out, fmt.Sprintf("%s\n", getTemporalView(tbl.Name, tbl.ViewColumns))); err != nil {
+				return err
+			}
+		}
 		for _, tbl := range tables {
-			if schemaOnly && tbl.TableType == baseTableType {
+			if tbl.TableType != baseTableType {
```

Comment:
add a comment before to suggest this is creating table

Reply:
Done.